simulation/main.go

```diff
@@ -0,0 +1,333 @@
+
+	log.Infof("Client wallets initialized")
+
+	for _, round := range simulation.Rounds {
+		log.Infof("Executing Round %d\n", round.Number)
+		var wg sync.WaitGroup
+		for clientID, actions := range round.Actions {
+			user, ok := users[clientID]
+			if !ok {
+				log.Fatalf("User %s not found", clientID)
+			}
+			for _, action := range actions {
+				wg.Add(1)
+				go func(u User, a interface{}) {
+					defer wg.Done()
+					actionMap := a.(map[string]interface{})
+					actionType := actionMap["type"].(string)
+					amount, _ := actionMap["amount"].(float64)
+					to, _ := actionMap["to"].(string)
+
+					var err error
+					switch actionType {
+					case "Onboard":
+						err = onboard(u, amount)
+					case "SendAsync":
+						err = sendAsync(u, amount, to, users)
+					case "Claim":
+						err = claim(u)
+					default:
+						log.Printf("Unknown action type: %s for user %s", actionType, u.ID)
+						return
+					}
+
+					if err != nil {
+						log.Printf("%s failed for user %s: %v", actionType, u.ID, err)
+					}
+				}(user, action)
+			}
+		}
+		wg.Wait()
+		time.Sleep(time.Duration(simulation.Server.RoundInterval) * 2 * time.Second)
+	}
+
+}
+
+func loadAndValidateSimulation(simFile string) (*Simulation, error) {
+	schemaBytes, err := os.ReadFile("schema.yaml")
+	if err != nil {
+		return nil, fmt.Errorf("error reading schema file: %v", err)
+	}
+
+	schemaJSON, err := yaml.YAMLToJSON(schemaBytes)
+	if err != nil {
+		return nil, fmt.Errorf("error converting schema YAML to JSON: %v", err)
+	}
+
+	schemaLoader := gojsonschema.NewBytesLoader(schemaJSON)
+
+	simBytes, err := os.ReadFile(simFile)
+	if err != nil {
+		return nil, fmt.Errorf("error reading simulation file: %v", err)
+	}
+
+	var sim Simulation
+	err = yaml.Unmarshal(simBytes, &sim)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing simulation YAML: %v", err)
+	}
+
+	simJSON, err := yaml.YAMLToJSON(simBytes)
+	if err != nil {
+		return nil, fmt.Errorf("error converting simulation YAML to JSON: %v", err)
+	}
+
+	documentLoader := gojsonschema.NewBytesLoader(simJSON)
+
+	result, err := gojsonschema.Validate(schemaLoader, documentLoader)
+	if err != nil {
+		return nil, fmt.Errorf("error validating simulation: %v", err)
+	}
+
+	if !result.Valid() {
+		// Collect error messages
+		var errorMessages string
+		for _, desc := range result.Errors() {
+			errorMessages += fmt.Sprintf("- %s\n", desc)
+		}
+		return nil, fmt.Errorf("the simulation is not valid:\n%s", errorMessages)
+	}
+
+	return &sim, nil
+}
+
+func setupArkClient() (arksdk.ArkClient, error) {
+	tempDir, err := os.MkdirTemp("", "ark_client_*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temporary directory: %s", err)
+	}
+
+	// Store the temporary directory path for later cleanup
+	tempDirsMutex.Lock()
+	tempDirs = append(tempDirs, tempDir)
+	tempDirsMutex.Unlock()
+
+	appDataStore, err := store.NewStore(store.Config{
+		ConfigStoreType:  types.FileStore,
+		AppDataStoreType: types.KVStore,
```

Comment:
we could use inmemory store and and prevent handling temp dirs, no need for appdata store as the clients are initialized without tx feed in client.Init():
```suggestion
		ConfigStoreType:  types.InMemoryStore,
```

Reply:
Done